src/miner.rs

```diff
@@ -113,6 +117,15 @@ impl Worker {
                     }
                     counter += 1;
 
+                    // if enough number of samples are collected
```

Comment:
Does this affect the hashrate of machine, given that it is run on every hash run? I feel the effect is small, but better check to make sure

Reply:
Yes, If workers are suspended for a reason, hash rate calculation will be restarted. If you are asking about `counter` and `512` limits, I didn't see any effect of them.

Reply:
No. i mean, is this optimized? I was worried that calculating hash-rate on each hash may reduce the overall hashrate of the machine

Reply:
If this is a concern, we need to implement another approach i.e. calculation hash rate in specific time frames and stop calculation, however, this is difficult and might Abe unstable. I don't know :) If `counter+=1` does not affect hash rate, then hash rate calculator won't affect much I suppose.